x-pack/plugins/maps/public/actions/map_actions.ts

```diff
@@ -369,8 +369,13 @@ export function addNewFeatureToIndex(geometry: Geometry | Position[]) {
     }
 
     try {
+      dispatch(updateEditShape(DRAW_SHAPE.WAIT));
       await (layer as IVectorLayer).addFeature(geometry);
-      await dispatch(syncDataForLayerDueToDrawing(layer));
+      await dispatch(syncDataForLayerDueToDrawing(layer)).then(() => {
+        setTimeout(() => {
+          dispatch(updateEditShape(DRAW_SHAPE.SIMPLE_SELECT));
+        }, 500);
```

Comment:
What is this timeout for?

Reply:
Animation for adding/removing shape on the map takes some fixed time, we need to return from wait state to the selected shape tool after it is over, it's not depends on the network performance.